[IMP] inventory: fix lot assignment (#9088)
odoo/documentation: content/applications/inventory_and_mrp/inventory/product_management/product_tracking/lots.rst

===========
Lot numbers
===========

.. |PO| replace:: :abbr:`PO (Purchase Order)`
.. |SO| replace:: :abbr:`SO (Sales Order)`
.. |DO| replace:: :abbr:`DO (Delivery Order)`
.. |cog| replace:: :icon:`fa-cog` :guilabel:`(cog)`
.. |list| replace:: :icon:`fa-list` :guilabel:`(list)`
.. |dropdown| replace:: :icon:`fa-caret-down` :guilabel:`(caret down)`

*Lots* are one of the two ways to identify and track products in Odoo. They typically represent a
specific batch of products that were received, stored, shipped, or manufactured in-house.

Manufacturers assign lot numbers to groups of products sharing common properties, facilitating
end-to-end traceability through their lifecycles.

Lots are useful for managing large quantities of manufactured or received products, aiding in
tracing items back to their group, particularly for product recalls or :doc:`expiration dates
<expiration_dates>`.

.. seealso::
   :doc:`serial_numbers`

Enable lots & serial numbers
============================

To track products using lots, enable the *Lots & Serial Numbers* feature. Go to the
:menuselection:`Inventory app --> Configuration --> Settings`, scroll down to the

> :guilabel:`Traceability` section, and tick the checkbox next to :guilabel:`Lots & Serial Numbers`.

Reviewer: Thanks for pointing this out! I added a seealso for this, below (:

> :guilabel:`Traceability` section, and tick the checkbox next to :guilabel:`Lots & Serial Numbers`.

Reviewer: I thought it was a little unrelated to this PR, but I also provided links to read more about it in the seealso admonition! :D

Then, click :guilabel:`Save`.

.. seealso::
   - :doc:`Tracking expiration dates <expiration_dates>`
   - :ref:`Print GS1 barcodes for lots and serial numbers <barcode/operations/gs1-lots>`

.. image:: lots/enabled-lots-setting.png
   :align: center
   :alt: Enabled lots and serial numbers feature in inventory settings.

.. _inventory/management/track_products_by_lots:

Track by lots
=============

Once the :guilabel:`Lots & Serial Numbers` feature is activated, configure individual products to be
tracked using lots. To do this, go to :menuselection:`Inventory app --> Products --> Products`, and
choose a product to configure.

On the product form, go to the :guilabel:`Inventory` tab. In the :guilabel:`Traceability` section,

> select the :guilabel:`By Lots` option in the :guilabel:`Tracking` field. Now, new or existing lot

Reviewer: added a seealso here (: thanks for the suggestion!

numbers can be assigned to newly-received or manufactured batches of this product.

.. important::

>    If a product has stock on-hand prior to activating tracking by lots or serial numbers, use an

Reviewer: This is helpful! I'll link to relevant documentation as well (:

   .. image:: lots/user-error.png
      :align: center
      :alt: Add lot/serial number user error popup.

On the receipt form, on the product line in the :guilabel:`Operations` tab, select the |list| icon
to the right of the product that is tracked by lot numbers.

.. image:: lots/list-icon.png
   :align: center
   :alt: Show the bulleted list icon on the product line.

Doing so opens the :guilabel:`Open: Stock move` pop-up window, where the :guilabel:`Lot/Serial
Number` and :guilabel:`Quantity` are assigned.

The two ways to assign lot numbers: **manually** and **importing**.

Manual assignment
~~~~~~~~~~~~~~~~~

To manually assign lot numbers, click :guilabel:`Add a line`. Input the :guilabel:`Lot/Serial
Number`, :guilabel:`Store To` location for the lot, :guilabel:`Quantity`, and :guilabel:`Destination
Package`, if any.

.. note::
   To assign multiple lot numbers, or store to multiple locations, click :guilabel:`Add a line`, and
   type a new :guilabel:`Lot/Serial Number` for additional quantities. Repeat until the total in the
   :guilabel:`Quantity` column matches the :guilabel:`Demand` at the top.

.. image:: lots/assign-lot-number-popup.png
   :align: center

>    :alt: Assign lot number detailed operations popup.

Reviewer: Can you tell me how you got to this pop-up? I went to PO > Receipt smart button > and clicked the numbered list icon. I wasn't able to find this popup, as mine looks like this
![assign-lots-popup](https://github.com/odoo/documentation/assets/23611711/8e205482-fcf2-452f-8f3a-a02888a49166)

.. seealso::
   :ref:`Traceability report for lot numbers <inventory/product_management/lot-traceability>`

.. _inventory/product_management/assign-lots-delivery:

On delivery orders
------------------

Odoo makes it possible to specify which lot numbers for a product are chosen for outgoing shipment
on a delivery order form.

To begin, create or select an existing quotation from the :menuselection:`Sales` app. After
confirming the |SO|, the :guilabel:`Delivery` smart button becomes available. Click the
:guilabel:`Delivery` smart button to view the warehouse receipt form for that specific |SO|.

.. note::
   Alternatively, navigate to delivery orders by going to the :menuselection:`Inventory` app, and
   clicking the :guilabel:`Delivery Orders` kanban card.

Clicking the :guilabel:`Delivery` smart button opens the the delivery order form, where lot numbers
are picked for delivery. In the :guilabel:`Operations` tab, click the |list| icon to the right of
the product that is tracked by lot numbers. Clicking that icon reveals a :guilabel:`Open: Stock
move` pop-up window.

In the pop-up window, the chosen lot number and its storage location is displayed in the

> :guilabel:`Pick From` column, with the with the full :guilabel:`Quantity` taken from that specific

Reviewer: Yes! this is helpful. Adding it now (:

going to :menuselection:`Inventory app --> Products --> Lots/Serial Numbers`.

By default, lot numbers are grouped by product, and selecting the drop-down menu for each product
displays the existing lot numbers. Select a lot number to :ref:`modify or add details
<inventory/product_management/edit-lot>` linked to the lot. Lot numbers can also be :ref:`created
<inventory/product_management/create-new-lot>` from this page, by clicking the :guilabel:`New`
button.

.. figure:: lots/lot-dashboard.png
   :align: center
   :alt: Show the "Lot/Serial Number" dashboard.

   Display lot numbers, grouped by products, on the **Lot/Serial Number** dashboard.

.. _inventory/product_management/edit-lot:

Modify lot
----------

Clicking a lot from the :guilabel:`Lot/Serial Number` dashboard reveals a separate page where
additional information can be provided about the lot.

.. tip::
   Odoo automatically generates a new :guilabel:`Lot/Serial Number` to follow the most recent
   number. However, it can be edited, by clicking the line under the :guilabel:`Lot/Serial Number`
   field, and changing the generated number to any desired one.

On the lot number form, the following fields can be modified:

> - :guilabel:`Lot/Serial Number`: Change the lot number linked to the :guilabel:`Product`

Reviewer: Thanks for pointing this out! i realize my note was vague. I meant that after a lot number was already created, the product cannot be modified

inventory adjustment.

.. example::
   After creating the lot number, `000001` appears as an option for `Drawer Black` when assigning
   lot numbers on the :guilabel:`Inventory Adjustment` page.

   .. image:: lots/inventory-adjustment.png
      :align: center
      :alt: Show how to assign lot numbers on the Inventory Adjustment page.

Manage lots for different operations types
==========================================

By default, new lots can only be created when receiving products, and existing lot numbers cannot
be used. For sales orders, only existing lot numbers can be utilized, and new ones cannot be created
on the delivery order.

To change the ability to use new (or existing) lot numbers on any operation type, go to the
:menuselection:`Inventory app --> Configuration --> Operations Types`, and select the desired
operation type.

On the operation type form, under the :guilabel:`Lots/Serial Numbers` section, tick the
:guilabel:`Create New` checkbox to enable new lot numbers to be created during this operation type.
Choose :guilabel:`Use Existing ones` if only existing lot numbers can be selected.

.. image:: lots/operation-type-form.png
   :align: center

>    :alt: Enabled traceability setting on operations type form.

Reviewer 2: This image is a little different from runbot:
![2024-05-06_12-10](https://github.com/odoo/documentation/assets/91896312/53260a7d-3a1c-48f3-9831-47ee71a74f6c)

Reviewer: Omg right, the font looks different somehow! I feel super confused. We're both in 17.0 runbot, but why does it look different?
![image](https://github.com/odoo/documentation/assets/23611711/d2d719e5-ee19-441d-a13c-675c551204d3)

   number. However, it can be edited, by clicking the line under the :guilabel:`Lot/Serial Number`
   field, and changing the generated number to any desired one.

On the lot number form, the following fields can be modified:

- :guilabel:`Lot/Serial Number`: change the lot number linked to the :guilabel:`Product`
- :guilabel:`Internal Reference`: records an alternative lot/serial number used within the warehouse
  that differs from the one used by the supplier manufacturer.
- :guilabel:`Company`: specify the company where the lot number is available.
- :guilabel:`Description`: add extra details about the lot or serial number in this text field.

.. important::
   On existing lots, the :guilabel:`Product` and :guilabel:`On Hand Quantity` fields **cannot** be
   modified, as the lot numbers are linked with existing stock moves.

.. image:: lots/lot-number.png
   :align: center
   :alt: Show the lot number form.

.. seealso::
   :doc:`Set expiration dates for lots <expiration_dates>`

Add property
~~~~~~~~~~~~

To add custom fields to lots for enhanced traceability, there are two methods of adding properties
on a lot number form:

> #. Click the :icon:`fa-cog` :guilabel:`(cog)` icon at the top-left of the page, then select
>    :guilabel:`Add Properties` from the drop-down menu.
> 

Reviewer 3: Please add icons to other buttons for added clarity :)
```suggestion
#. Click the :icon:`fa-cog` :guilabel:`(cog)` icon at the top-left of the page, then select
   :icon:`fa-cogs` :guilabel:`Add Properties` from the drop-down menu.
```

> #. Click the :guilabel:`Add a Property` button, located below the existing fields.

Reviewer 3: ```suggestion
#. Click the :icon:`fa-plus` :guilabel:`Add a Property` button, located below the existing fields.
```

From here, products with lot numbers assigned to them will be listed by default, and can be expanded
to show the lot numbers those products have assigned to them.

To group by lots, begin by removing any filters in the :guilabel:`Search...` bar. Then, click the
|dropdown| icon to open a drop-down menu of filters. Click the :guilabel:`Add Custom Group` option,

> and select :guilabel:`Lot/Serial Number` from the drop-down menu.

Reviewer 2: this step is unclear as there are more options available

From here, products with lot numbers assigned to them will be listed by default, and can be expanded
to show the lot numbers those products have assigned to them.

To group by lots, begin by removing any filters in the :guilabel:`Search...` bar. Then, click the

> |dropdown| icon to open a drop-down menu of filters, groupbys, and favorites. Under the

Reviewer 4: ```suggestion
|dropdown| icon to open a drop-down menu of :guilabel:`Filters`, :guilabel:`Group By` options, and :guilabel:`Favorites`. Under the
```
